### PR TITLE
feature/product listing

### PR DESCRIPTION
**Pull request by thegeorgenikhil**
Made a page for listing all the game cards
- added context for listing page
- **load data** from the mock backend on the first load
- sort games by **price**
- filter games by **category**
- filter games by **rating**
- filter games by **price**

[Preview Here](https://deploy-preview-2--gamesandbox.netlify.app/games/all)

### CODE REVIEW

**Comment by ravikumar1002** [comment]
Hey Nikhil, your UI looks amazing and your filter section is on the right side which is unique
   some suggestions:
    * Add Clear Filter Button
    * I think your range feature is working differently because products will typically show below that amount

File: src/pages/AllGames.jsx [hunk 1]
```diff
@@ -28,34 +80,65 @@ const AllGames = () => {
...
+                </div>
+              </div>
+              <div className="divider-sm"></div>
+
               <div className="filter-group">
                 <p className="filter-heading">CATEGORY</p>
                 <div className="filter-input-container">
                   <div className="filter-input-group">
-                    <input type="checkbox" name="racing" />
+                    <input
```
Comment (ravikumar1002) [inline src/pages/AllGames.jsx:121]: One suggestion, you can put this input in the label of category, it will look good

Comment (ravikumar1002) [inline src/pages/AllGames.jsx:133]: I think you forgot to add id here

**Review by ravikumar1002 — APPROVED** [review]
Looks good to merge
And step(input) attribute was new to me
folder structure is good 👍

**Comment by thegeorgenikhil** [comment]
> Hey Nikhil, your UI looks amazing and your filter section is on the right side which is unique some suggestions: * Add Clear Filter Button * I think your range feature is working differently because products will typically show below that amount

Yeah, it should start from the max value, I got it reversed. Thank you for pointing it out. Will correct that👍

**Review by rahulbakshi4 — APPROVED** [review]
- Code looks clean
- Nice use of helper functions
- Good you used cloudinary for image hosting 

As per the preview you range slider works the opposite way  fix that 

Good to merge

**Pull request merged**